### PR TITLE
Miscellaneous ILM cleanups

### PR DESCRIPTION
**Pull request by joegallo**
This is an unconnected bunch of commits that are just tidying some little things here and there that I ran into in the ILM code while I was working on my current task.

This is a continuation of #118466 and #118338, there's also a little bit of #118354 in there, too.

### CODE REVIEW

File: x-pack/plugin/core/src/test/java/org/elasticsearch/xpack/core/ilm/LifecyclePolicyTests.java [hunk 1]
```diff
@@ -151,17 +151,7 @@ public static LifecyclePolicy randomTimeseriesLifecyclePolicy(@Nullable String l
             // Remove the frozen phase, we'll randomly re-add it later
             .filter(pn -> TimeseriesLifecycleType.FROZEN_PHASE.equals(pn) == false)
             .collect(Collectors.toList());
-        Map<String, Phase> phases = Maps.newMapWithExpectedSize(phaseNames.size());
-        Function<String, Set<String>> validActions = getPhaseToValidActions();
-        Function<String, LifecycleAction> randomAction = getNameToActionFunction();
-        // as what actions end up in the hot phase influence what actions are allowed in the subsequent phases we'll move the hot phase
-        // at the front of the phases to process (if it exists)
-        if (phaseNames.contains(TimeseriesLifecycleType.HOT_PHASE)) {
```
Comment (masseyke) [inline x-pack/plugin/core/src/test/java/org/elasticsearch/xpack/core/ilm/LifecyclePolicyTests.java:159]: Do we not need to worry about HOT_PHASE coming first any more? I don't really understand why it was necessary in the first place.

File: x-pack/plugin/ilm/src/main/java/org/elasticsearch/xpack/ilm/PolicyStepsRegistry.java [hunk 2]
```diff
@@ -269,9 +269,8 @@ public Set<Step.StepKey> parseStepKeysFromPhase(String policy, String currentPha
             return parseStepsFromPhase(policy, currentPhase, phaseDefNonNull).stream().map(Step::getKey).collect(Collectors.toSet());
         } catch (IOException e) {
             logger.trace(
-                () -> String.format(
-                    Locale.ROOT,
-                    "unable to parse steps for policy [{}], phase [{}], and phase definition [{}]",
```
Comment (masseyke) [inline x-pack/plugin/ilm/src/main/java/org/elasticsearch/xpack/ilm/PolicyStepsRegistry.java:274]: Oh nice -- we pick up a bug fix!

File: x-pack/plugin/ilm/src/test/java/org/elasticsearch/xpack/ilm/IndexLifecycleTransitionTests.java [hunk 3]
```diff
@@ -1436,6 +1436,6 @@ private void assertClusterStateStepInfo(
         assertEquals(expectedstepInfoValue, newLifecycleState.stepInfo());
         assertEquals(oldLifecycleState.phaseTime(), newLifecycleState.phaseTime());
         assertEquals(oldLifecycleState.actionTime(), newLifecycleState.actionTime());
-        assertEquals(newLifecycleState.stepTime(), newLifecycleState.stepTime());
```
Comment (masseyke) [inline x-pack/plugin/ilm/src/test/java/org/elasticsearch/xpack/ilm/IndexLifecycleTransitionTests.java:1439]: Nice, another (test) bug fix.

**Review by masseyke — APPROVED** [review]
LGTM, although I don't understand the one block of test code you removed (or why it was there in the first place).

File: x-pack/plugin/core/src/test/java/org/elasticsearch/xpack/core/ilm/LifecyclePolicyTests.java (hunk 1 above)
Comment (joegallo) [reply x-pack/plugin/core/src/test/java/org/elasticsearch/xpack/core/ilm/LifecyclePolicyTests.java:159]: My read of it is that this code was maybe necessary **once**, but the loop after this that I didn't remove ended up making this code redundant anyway. https://github.com/elastic/elasticsearch/pull/118488/files/93a7c08283627686d9ef83f7e80793be004c8bb8#diff-ea4f8cb2981d1e363c5308fc815b3ce4f94ec116f7e9691d69bbe63d96c607a4R155-R161

Comment (joegallo) [reply x-pack/plugin/core/src/test/java/org/elasticsearch/xpack/core/ilm/LifecyclePolicyTests.java:159]: Yeah, that tracks the history -- the first (now redundant) block that I've removed was added in #64883, while the second was added a few months later in #68864.

File: x-pack/plugin/ilm/src/main/java/org/elasticsearch/xpack/ilm/PolicyStepsRegistry.java (hunk 2 above)
Comment (joegallo) [reply x-pack/plugin/ilm/src/main/java/org/elasticsearch/xpack/ilm/PolicyStepsRegistry.java:274]: Yeah, I'm just annoyed I didn't catch this on the first go around!

**Pull request merged**